src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -161,10 +167,11 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
      * Parses block comment content as javadoc comment.
      * @param blockComment
      *        block comment content.
+     * @param result Reference to object returned by {@link #parseJavadocAsDetailNode(DetailAST)}
      * @return parse tree
      * @noinspection deprecation
      */
-    private ParseTree parseJavadocAsParseTree(String blockComment) {
+    private ParseTree parseJavadocAsParseTree(String blockComment, ParseStatus result) {
```

Comment:
We can change
 1. `parseJavadocAsParseTree` to `createJavadocParser` and do all other processing in `parseJavadocAsDetailNode`

2. Make `result` or `firstNonTightHtmlTagToken` a field of the class `JavadocDetailNodeParser`